pwiz_tools/Skyline/CommandArgUsage.resx

```diff
@@ -1021,4 +1021,37 @@ Useful in sample mixtures including multiple species.</value>
   <data name="_associate_proteins_gene_level_parsimony" xml:space="preserve">
     <value>Associate peptides with genes (or gene groups) instead of proteins, and apply parsimony options to that association.</value>
   </data>
+  <data name="CommandArgs_MOD_NAME_VALUE_full_name__e_g___Oxidation__M______short_name__Oxi_" xml:space="preserve">
+    <value>full name (e.g. "Oxidation (M)") | short name (Oxi)</value>
+  </data>
+  <data name="CommandArgs_MOD_UNIMOD_VALUE_UniMod_ID__e_g___35__" xml:space="preserve">
+    <value>UniMod ID (e.g. "35")</value>
+  </data>
+  <data name="_integrate_all" xml:space="preserve">
+    <value>When set all transitions contributing any signal between the integration boundaries are considered found. When not set, only transitions that are coeluting are considered found. Regardless of the setting all signal from the quantitative transitions between the integration boundaries is summed to produce the TotalArea for the precursor.</value>
+  </data>
+  <data name="_pep_add_mod" xml:space="preserve">
+    <value>Add a UniMod modification to the document. The modification can be specified by long name ("Oxidation (M)") or short name ("Oxi"). If using the short name, you may have to use --pep-add-mod-aa or --pep-add-mod-term to choose which specificity of the modification that you want to add to the document.</value>
+  </data>
+  <data name="_pep_add_mod_aa" xml:space="preserve">
+    <value>Sets the amino acid specificity of the last peptide modification added by --pep-add-mod.</value>
```

Comment:
I feel "Specifies" is the most technically appropriate word but "Specifies the specificity" is cringy. I'm fine changing to "Provides" but I went with "Sets" in the sense of: "You are setting the specificity of the previously given mod name/id".

To be honest I'd rather get rid of the ambiguous mod arguments and require the long form all the time. I went ahead and implemented it all so you can see what it's like, but I think it's clumsy. Any time that a command-line depends on the order of --key=value arguments, it's awkward. I suppose that's why every major command-line parsing framework handles "positional" arguments differently than --key=value ones.

`--pep-add-mod` does not currently work with "Oxidation" - the name must exactly match either the ShortName or Name property in UniModData. I agree that's not ideal because the UniModData names don't actually match to any single field of UniMod! Users can't find either `Oxidation (M)` OR `Oxi` at UniMod's page for oxidation:
https://unimod.org/modifications_view.php?editid1=35

That's because Skyline's concept of UniMod uses Mascot names, right? (And short names from ProteinPilot.) And the Mascot name (mostly?) corresponds to the PSI-MS Name field in UniMod, plus the specificity. **So perhaps the --pep-add-mod description should make it clear that the "long name" is defined as the PSI-MS Name plus the specificity in parentheses?** And I'm fine with adding the PSI-MS Name (StaticMod Name stripped of the parentheses) as an alternate to the ambiguous short-name lookup if you want to keep the ambiguous support.

Reply:
Should we add a menu item "Help > Documentation > Unimod modifications" which would bring up a window showing all of the unimod modifications that Skyline knows about? It would show the chemical formula, short name, specificity, etc.
Would that make it easier for users to figure out what they should call a modification on the commandline?

Reply:
Sounds like a good idea. I might make it accessible by a new detailed command-line usage flag (something like --help-mods) so users don't have to open the GUI to access it, though, yet aren't bombarded with it with a basic --help call. Yet it could still be accessed in the GUI, just like the other command-line usage is. Probably for another PR though. For this one I need to know whether Brendan wants to keep the ambiguous mod support (and extend it for long name matches without specificity).

Reply:
Mustn't there be a lot of modifications that have only one specificity? I like the idea of the documentation. In the UI version, we could make the Unimod IDs links to the page on the Unimod website. Like the one above for Oxidation.

Let's drop the SCIEX short names, and only support the full name in --pep-add-mod alone, or --pep-add-unimod alone or with --pep-add-unimod-aa and --pep-add-unimod-term. I don't see that we need to teach people the SCIEX short names, and we don't really have a good reference for them like we do for Unimod IDs.

Reply:
i.e. Let's cut the feature down instead of expanding it to handle ambiguous full names.

Reply:
And to be clear, by full name you mean "Oxidation (M)" not "Oxidation"?